scripts/upload_demo_to_space.py

```diff
@@ -1,78 +1,159 @@
+        gradio_version (Optional[str]): If provided, sets the Gradio version in the created space.
+
+    Returns:
+        str: URL of the uploaded Hugging Face Space.
     """
 
+    print(f"Uploading demo '{demo_name}' to space '{space_id}'...")
+
+    def split_into_chunks(lst: List, n: int) -> List[List]:
+        """
+        Splits a list into chunks of size n.
+
+        Args:
+            lst (List): The list to split.
+            n (int): The size of each chunk.
+
+        Returns:
+            List[List]: A list of chunks.
+        """
+        for i in range(0, len(lst), n):
+            yield lst[i : i + n]
+
     with tempfile.TemporaryDirectory() as tmpdir:
-        demo_path = pathlib.Path(pathlib.Path().absolute(), f"demo/{demo_name}")
+        demo_path = pathlib.Path.cwd() / "demo" / demo_name
+        if not demo_path.exists():
+            raise FileNotFoundError(f"Demo path '{demo_path}' does not exist.")
+
+        # Copy demo directory to temporary directory
         shutil.copytree(demo_path, tmpdir, dirs_exist_ok=True)
+
+        # Rename 'run.py' to 'app.py'
         app_file = pathlib.Path(tmpdir, "run.py")
-        # Rename the app file to be app.py
-        app_file.rename(app_file.with_stem("app"))
+        if app_file.exists():
+            app_file.rename(app_file.with_stem("app"))
+        else:
+            raise FileNotFoundError(
+                f"App file '{app_file}' does not exist in the demo directory."
+            )
+
+        # Update README.md with Gradio version if provided
         if gradio_version:
             readme = pathlib.Path(tmpdir, "README.md")
             readme_content = f"""
-                                ---
-                                title: {space_id.split("/")[-1]} 
-                                emoji: 💩
-                                colorFrom: indigo
-                                colorTo: indigo
-                                sdk: gradio
-                                sdk_version: {gradio_version}
-                                app_file: app.py
-                                pinned: false
-                                ---
-                                """
-            readme.open("w").write(textwrap.dedent(readme_content))
-
-        api = huggingface_hub.HfApi()
-        huggingface_hub.create_repo(
-            space_id,
+            ---
+            title: {space_id.split("/")[-1]} 
+            emoji: 💩
+            colorFrom: indigo
+            colorTo: indigo
+            sdk: gradio
+            sdk_version: {gradio_version}
+            app_file: app.py
+            pinned: false
+            ---
+            """
+            readme.write_text(textwrap.dedent(readme_content))
+
+        print("Uploading files to Hugging Face Space...")
+        api = HfApi()
+
+        print("Creating repository...")
+
+        # Create the repository if it doesn't exist
+        api.create_repo(
```

Comment:
(nit) Here you could do `space_url = api.create_repo(...)` and then `space_id = space_url.repo_id`. This way the script will also work with implicit repo_id when the username is not provided. If you move the create_repo call a bit up, you can reuse `space_id` (the correct one with explicit namespace) in the logs + model card.

Reply:
Nice. Ty.